sdk/iot/hub/samples/telemetry/src/paho_iot_hub_telemetry_example.c

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifdef _MSC_VER
+// warning C4201: nonstandard extension used: nameless struct/union
+#pragma warning(disable : 4201)
+#endif
+#include <paho-mqtt/MQTTClient.h>
+#ifdef _MSC_VER
+#pragma warning(default : 4201)
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifdef _WIN32
+// Required for Sleep(DWORD)
+#include <Windows.h>
+#else
+// Required for sleep(unsigned int)
+#include <unistd.h>
+#endif
+
+#include <az_iot_hub_client.h>
+#include <az_result.h>
+#include <az_span.h>
+
+// TODO: #564 - Remove the use of the _az_cfh.h header in samples.
+//              Note: this is required to work-around MQTTClient.h as well as az_span init issues.
+#include <_az_cfg.h>
+
+// Device ID
+#define DEVICE_ID "AZ_IOT_DEVICE_ID"
+
+// IoT Hub FQDN
+#define IOT_HUB_FQDN "AZ_IOT_HUB_FQDN"
+
+// Device information
+#define REGISTRATION_ID_ENV "AZ_IOT_REGISTRATION_ID"
+
+// AZ_IOT_DEVICE_X509_CERT_PEM_FILE is the path to a PEM file containing the device certificate and
+// key as well as any intermediate certificates chaining to an uploaded group certificate.
+#define DEVICE_X509_CERT_PEM_FILE "AZ_IOT_DEVICE_X509_CERT_PEM_FILE"
+
+// AZ_IOT_DEVICE_X509_TRUST_PEM_FILE is the path to a PEM file containing the server trusted CA
+// This is usually not needed on Linux or Mac but needs to be set on Windows.
+#define DEVICE_X509_TRUST_PEM_FILE "AZ_IOT_DEVICE_X509_TRUST_PEM_FILE"
+
+#define NUMBER_OF_MESSAGES 5
+
+static char device_id[64];
+static char iot_hub_fqdn[128];
+static char x509_cert_pem_file[512];
+static char x509_trust_pem_file[256];
+char telemetry_topic[128];
+
+static char paho_endpoint[128];
+static az_span paho_prefix = AZ_SPAN_LITERAL_FROM_STR("ssl://");
+static az_span paho_suffix = AZ_SPAN_LITERAL_FROM_STR(":8883");
+
+static const char* telemetry_message_payloads[NUMBER_OF_MESSAGES] = {
+  "Message One", "Message Two", "Message Three", "Message Four", "Message Five",
```

Comment:
should we make these JSON payloads?

Reply:
Here's a thought: don't put any JSON-speciifc things in your client library code if you think you might offer other protocols (like CBOR) in the future. The "fear" is that you'll carry data/code in your library that will be useless for a CBOR customer.

Reply:
telemetry doesn't need to be JSON so it isn't here. Things like methods responses do need to be in JSON so they are.

Reply:
Will your statement remain true if CBOR becomes a customer choice?

Reply:
Not necessarily. But in that case we can change these samples to included CBOR capabilities. Something like having a `generate_payload()` and having it do JSON or CBOR. For now, only the features that fundamentally require JSON like DPS have it internally. Otherwise, it is up to the user to create the correct payloads, in the necessary encoding. Example:
https://github.com/Azure/azure-sdk-for-c/blob/1848fd042adceedc2c3c0202c3ceb228435c33d4/sdk/iot/hub/samples/twin/src/iot_hub_twin_example.c#L306-L318
In ⬆️ case we can just as easily do this encoding in CBOR using [tinyCBOR](https://github.com/intel/tinycbor)

Reply:
I don't think 99% of devices will report variable sized data (e.g. sensors are normally 4-8 bytes). They don't need to be descriptive because IOT Hub already knows which device is sending data and so it can be as compressed as possible (i.e. send only the actual binary values, perform adaptation based on device capabilities on the service side where battery and CPU power are not a concern).